packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -306,7 +306,19 @@ class ShortcutManager extends ChangeNotifier with Diagnosticable {
       return false;
     }
     assert(context != null);
-    final LogicalKeySet keySet = keysPressed ?? LogicalKeySet.fromSet(RawKeyboard.instance.keysPressed);
+    LogicalKeySet keySet = keysPressed;
+    if (keySet == null) {
+      assert(RawKeyboard.instance.keysPressed.isNotEmpty,
+        'Received a key down event when no keys are in keysPressed. '
+        "This state can occur if the key event being sent doesn't properly "
+        'set its modifier flags. This was the event: $event and its data: '
+        '${event.data}');
+      if (RawKeyboard.instance.keysPressed.isNotEmpty) {
```

Comment:
Do you mean this should crash in debug mode but pass in release mode? I kind of get the intention, but still why would anyone release a piece of code that crashes in debug mode?

Reply:
The reason is that unless they press the key that causes the problem during their testing, they won't know that it crashes in debug mode.

And I don't think an invalid key sequence should crash the app in release mode: it can be ignored if it gets that far.

Reply:
Sounds good :)

Reply:
Maybe you can add this to the comment. Your explanation is pretty helpful.

Reply:
OK, sounds good, added a comment.